gui/view/plot_diagram.cpp

```diff
@@ -70,10 +70,30 @@ void PlotDiagram::refresh()
     yAxis->setRange(rgeY.min, rgeY.max);
     xAxis->setVisible(true);
     yAxis->setVisible(true);
-    if (ysSigma.size() > 0) // has valueError
-        graph_->setDataValueError(QVector<double>::fromStdVector(xs), QVector<double>::fromStdVector(ys), QVector<double>::fromStdVector(ysSigma));
-    else
-        graph_->setData(QVector<double>::fromStdVector(xs), QVector<double>::fromStdVector(ys));
+    if (ysSigma.size() > 0) {// has valueError
+        std::vector<double> xsSafe;
+        std::vector<double> ysSafe;
+        std::vector<double> ysSigmaSafe;
+        for (size_t i = 0; i < xs.size(); ++i)
+            if (   !qIsNaN(xs.at(i)) && qIsFinite(xs.at(i))
+                && !qIsNaN(ys.at(i)) && qIsFinite(ys.at(i))) {
```

Comment:
bitte anders herum:

for (...) {
   if (anomaler Zustand)
      continue;
   normale Prozedur;
}